src/utils/generate_cxx_code.cpp

```diff
@@ -659,84 +967,60 @@ const std::string  wcs::generate_cxx_code::generate_code(const LIBSBML_CPP_NAMES
+
+      for (unsigned int ic = 0u; ic < eventsSize; ic++) {
+        const LIBSBML_CPP_NAMESPACE::Event& event = *(events_list->get(ic));
+        const ListOfEventAssignments* event_assignment_list
+        = event.getListOfEventAssignments();
+        const unsigned int num_event_assignments = event_assignment_list->size();
+        const Trigger* trigger = event.getTrigger();
+        const ASTNode* astnode = trigger->getMath();
+        std::string trigger_variable = SBML_formulaToString(astnode->getChild(0));
+        std::string trigger_value = SBML_formulaToString(astnode->getChild(1));
+        LIBSBML_CPP_NAMESPACE::ASTNode math = *astnode->getChild(1);
+        update_scope_ast_node(math, wcs_all_var, wcs_all_const, {});
+        for (unsigned int ici = 0u; ici < num_event_assignments; ici++) {
+          const LIBSBML_CPP_NAMESPACE::EventAssignment& eventassignment
+          = *(event_assignment_list->get(ici));
+          if (eventassignment.getVariable() == x){
+            //genfile << "  if (variable_tr_ev == \"" << trigger_variable << "\" && value_tr_ev == " << trigger_value << ") {\n";
+            genfile << "  if (time == " << SBML_formulaToString(&math) << ") {\n";
+            genfile << "    " << x << " = " << SBML_formulaToString(eventassignment.getMath()) << ";\n";
+            genfile << "    wcs_global_var." << x << " = " << x << ";\n";
+            genfile << "  }\n";
+          }
+        }
       }
-    }
-    std::string formula = SBML_formulaToString(reaction.getKineticLaw()->getMath());
-    const auto input_map = build_input_map(formula, var_names);
 
-    std::vector<std::string> var_names_ord (input_map.size());
-    for (auto& x: input_map) {
-      var_names_ord[x.second] = x.first;
+      genfile << "\n  return " << x << ";\n" << "}\n\n";
     }
+  }
 
-    //print first parameters in formula taking input
-    std::vector<std::string>::iterator it;
-    for (it = var_names_ord.begin(); it < var_names_ord.end(); it++){
-      genfile << "  " << Real << " " << *it <<" = __input[__ii++];\n";
-      var_names.erase(*it);
-    }
-    //print rest parameters (not in formula) taking input
-    for (auto& x: var_names) {
-      genfile << "  " << Real << " " << x <<" = __input[__ii++];\n";
-    }
 
-    //print the rest of the parameters not taking an input
-    for (auto it = dependencies_set.crbegin();
-         it != dependencies_set.crend(); ++it)
-    {
-      arit = assignment_rules_set.find(*it);
-      if (arit != assignment_rules_set.cend()){
-        genfile << "  " << Real << " " << arit->first << " = "
-                  << SBML_formulaToString(arit->second) << ";\n";
-      }
-    }
 
-    genfile << "  " << Real << " " << reaction.getIdAttribute() << " = "
-              << SBML_formulaToString(reaction.getKineticLaw()->getMath())
-              << ";\n";
-    genfile << "  return " << reaction.getIdAttribute() << ";\n" << "}\n\n";
-  }
 
   genfile << "\n//Define differential rates\n";
```

Comment:
line 570 and this contains the words 'differential rates'
This probably be something else more general since we do not do this for ODE.

Reply:
I put "//Define the functions for updating global state variables" instead of this. I hope it is okay.